sklearn/feature_extraction/text.py

```diff
@@ -2011,10 +1975,25 @@ def idf_(self):
         -------
         ndarray of shape (n_features,)
         """
+        if not hasattr(self, "_tfidf"):
+            raise NotFittedError(
+                f"{self.__class__.__name__} is not fitted yet. Call 'fit' with "
+                "appropriate arguments before using this attribute."
+            )
         return self._tfidf.idf_
 
     @idf_.setter
     def idf_(self, value):
+        if not hasattr(self, "_tfidf"):
+            # We should support transfering `idf_` from another `TfidfTransformer`
+            # and therefore, we need to create the transformer instance it does not
+            # exist yet.
+            self._tfidf = TfidfTransformer(
+                norm=self.norm,
+                use_idf=self.use_idf,
+                smooth_idf=self.smooth_idf,
+                sublinear_tf=self.sublinear_tf,
+            )
```

Comment:
I don't get the motivation for setting fitted attributes this way. How would this break without this setter?

Reply:
The following test would be broken:

https://github.com/scikit-learn/scikit-learn/blob/384661fbb39ec6e4476c6f085d019ee3b51056fd/sklearn/feature_extraction/tests/test_text.py#L1221-L1229

because `estimator._tfidf` would not exist.

Reply:
It looks like we wanted to support such transfer of the `idf_` vector.

Reply:
Indeed, thanks for the clarification.

Reply:
The way this mechanism works goes against scikit-learn's API. `transform` ends setting `vocabulary_` and `fixed_vocabulary_`:

https://github.com/scikit-learn/scikit-learn/blob/80ebe21ec280892df98a02d8fdd61cbf3988ccd6/sklearn/feature_extraction/text.py#L493-L499

If we want to follow scikit-learn's API  and transfer `idf`, we can have this API:

```python
orig = TfidfVectorizer(use_idf=True) 
orig.fit(JUNK_FOOD_DOCS) 
copy = TfidfVectorizer(vocabulary=orig.vocabulary_, use_idf=orig.idf_) 

assert_array_equal( 
    copy.fit_transform(JUNK_FOOD_DOCS).toarray(), 
    orig.transform(JUNK_FOOD_DOCS).toarray(), 
) 
```

That being said, I am happy with this PR as is.